distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/JvmErgonomics.java

```diff
@@ -55,6 +61,22 @@ static List<String> choose(final List<String> userDefinedJvmOptions) throws Inte
             ergonomicChoices.add("-XX:G1ReservePercent=" + tuneG1GCReservePercent);
         }
 
+        boolean isSearchTier = NodeRoleSettings.NODE_ROLES_SETTING.get(nodeSettings).contains(DiscoveryNodeRole.SEARCH_ROLE);
+        // override new percentage on small heaps on search tier to increase chance of staying free of the real memory circuit breaker limit
+        if (isSearchTier && heapSize < ByteSizeUnit.GB.toBytes(5)) {
+            ergonomicChoices.add("-XX:+UnlockExperimentalVMOptions");
+            ergonomicChoices.add("-XX:G1NewSizePercent=10");
+        }
+
+        // for dedicated search, using just 1 conc gc thread is not always enough to keep us below real memory breaker limit
+        // jvm use (2+processsors) / 4 (for small processor counts), so only affects 4/5 processors (for now)
+        if (EsExecutors.NODE_PROCESSORS_SETTING.exists(nodeSettings)) {
+            int allocated = EsExecutors.allocatedProcessors(nodeSettings);
+            if (allocated >= 4 && allocated <= 5 && isSearchTier) {
```

Comment:
`(allocated == 4 || allocated == 5) && isSearchTier` is clearer?

Reply:
I had that first, but preferred the above option, since it does not have both `||` and `&&`, but happy to adjust if you prefer.

Reply:
Minor point - up to you